Sources/SwiftIfConfig/BuildConfiguration.swift

```diff
@@ -0,0 +1,275 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+/// Describes the ordering of a sequence of bytes that make up a word of
+/// storage for a particular architecture.
+public enum Endianness: String {
+  /// Little endian, meaning that the least significant byte of a word is
+  /// stored at the lowest address.
+  case little
+
+  /// Big endian, meaning that the most significant byte of a word is stored
+  /// at the lowest address.
+  case big
+}
+
+/// Describes the requested version of a module.
+public enum CanImportVersion {
+  /// Any version of the module will suffice.
+  case unversioned
+
+  /// Only modules with the given version or higher will match.
+  case version(VersionTuple)
+
+  /// Only modules where the underlying Clang module has the given version or
+  /// higher will match.
+  case underlyingVersion(VersionTuple)
```

Comment:
IIRC `canImport` also supports build numbers, like `"20A301"`, so I don’t think we can use a `VersionTuple` here.

https://github.com/swiftlang/swift-syntax/blob/248dcef04d9e03b7fc47905a81fc84c6f6c23837/Tests/SwiftParserTest/translated/IfconfigExprTests.swift#L411-L420

Reply:
WAT

Reply:
Oh, these are actually errors in the old parser as well:

#if canImport(A, _version: 20A301) // expected-error {{'A' is not a valid digit in integer literal}}
  let a = 1
#endif

#if canImport(A, _version: "20A301") // expected-error {{cannot parse module version '20A301'}}
  let a = 1
#endif

I've added tests to https://github.com/swiftlang/swift-syntax/pull/2770

Reply:
Ah, I was slightly misremembering things. What is valid and what I forgot before is that the version number can be inside a string literal. (rdar://121070235)

```swift
canImport(MyLibrary, _version: "1.2.3")
```

Reply:
It's already supported. There's a test like this:

```
    assertIfConfig(#"canImport(SwiftSyntax, _version: "5.9")"#, .active)
```

Reply:
Oh, I missed that. Great!